package/windows/execute.ps1

```diff
@@ -286,9 +286,22 @@ if ($CATTLE_CA_CHECKSUM)
     $temp.MoveTo("$sslCertDir\serverca")
 
     # import the self-signed certificate
-    certoc.exe -addstore root "$sslCertDir\serverca" | Out-Null
-    if (-not $?) {
-        Log-Error "Failed to import rancher server certificates to Root"
+    $caBytes = $null
+    Get-Content "$sslCertDir\serverca" | % {
+        if ($_ -match '-+BEGIN CERTIFICATE-+') {
+            $caBytes = @()
+        } elseif ($_ -match '-+END CERTIFICATE-+') {
+            $caTemp = New-TemporaryFile
+            Set-Content -Value $caBytes -Path $caTemp.FullName -Encoding Byte
+            certoc.exe -addstore root $caTemp.FullName | Out-Null
+            if (-not $?) {
+                $caTemp.Delete()
+                Log-Error "Failed to import rancher server certificates to Root"
+            }
+            $caTemp.Delete()
```

Comment:
Looks like it won't exit on Log-Error: https://github.com/rancher/rancher/blob/883c41e029a70a482f612f569983ae86d84ab0bf/package/windows/utils.psm1#L19
Will it do delete twice when an error occurs?

Reply:
Using `Log-Fatal` to make the logic exit when failed, so the user could get noticed. `$caTemp.Delete()` just drops the temp file, those imported certs would store in `Cert:\LocalMachine\Root\`. If getting an import error on the last one of the file, I think we don't need to delete those certs already imported.